pinot-controller/src/test/java/org/apache/pinot/controller/api/PinotInstanceRestletResourceTest.java

```diff
@@ -54,9 +57,17 @@ public void testInstanceListingAndCreation()
       throws Exception {
     // Check that there is only one CONTROLLER instance in the cluster
     String listInstancesUrl = _controllerRequestURLBuilder.forInstanceList();
-    JsonNode instanceList = JsonUtils.stringToJsonNode(sendGetRequest(listInstancesUrl));
-    assertEquals(instanceList.get("instances").size(), 1);
-    assertTrue(instanceList.get("instances").get(0).asText().startsWith(Helix.PREFIX_OF_CONTROLLER_INSTANCE));
+
+    TestUtils.waitForCondition(aVoid -> {
+      try {
+        String getResponse = sendGetRequest(listInstancesUrl);
+        JsonNode jsonNode = JsonUtils.stringToJsonNode(getResponse);
+        return (jsonNode != null) && (jsonNode.get("instances") != null) && (jsonNode.get("instances").size() == 1)
+            && (jsonNode.get("instances").get(0).asText().startsWith(Helix.PREFIX_OF_CONTROLLER_INSTANCE));
+      } catch (Exception e) {
+        throw new RuntimeException(e);
+      }
+    }, 500L, 10000L, "Expected one controller instance");
```

Comment:
Another implementation of waitForCond () calls every 100ms. I think that should work fine here. We can still have the total time spent as 10s. Please declare that as a static final int and use it everywhere.thanks

Reply:
done